src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

```diff
@@ -340,6 +340,24 @@ public void ExistsDroppedExportProcessorItems(string exportProcessorName, string
             this.WriteEvent(32, exportProcessorName, exporterName, droppedCount);
         }
 
+        [Event(33, Message = "Measurements from instrument '{0}' is ignored as maximum metric streams limit reached.", Level = EventLevel.Warning)]
```

Comment:
Curious, do we need these separate events, or one general event would work fine? (my worry is that we keep adding events and we don't know how to manage them)

Reply:
would be good to have separate event for instrument drop, for each unique reason. i.e dropped due to max Metric limit and dropped due to conflict are separate events.

Reply:
I wish to understand more about this, will discuss with you offline.

Reply:
Removed the additional logging from this PR. I intend to bring it back after most Metric functionalities are in good shape, Need to also consider the original proposal about self-diagnostics https://github.com/open-telemetry/opentelemetry-dotnet/issues/1258.
It might be good enough to use generic Log messages for majority of the cases.
For important things like telemetry loss (for variety is reasons like cap, invalid name, conflict, unsubscribed source etc.), we need a dedicated Event, with "reason" as one of the parameter.
eg:
LogInstrumentIgnored(instrumentname, metername, reason)
LogMeasurementIgnored(instrumentname,metername,metricname,reason)